tests/components/august/test_init.py

```diff
@@ -1,136 +1,146 @@
-    assert data.lock_has_doorsense("doorsenselock1") is False
 
+async def test_lock_has_doorsense(hass):
+    """Check to see if a lock has doorsense."""
+    doorsenselock = await _mock_doorsense_enabled_august_lock_detail(hass)
+    nodoorsenselock = await _mock_doorsense_missing_august_lock_detail(hass)
+    await _create_august_with_devices(hass, [doorsenselock, nodoorsenselock])
 
-async def test__refresh_access_token(hass):
-    """Test refresh of the access token."""
-    authentication = _mock_august_authentication("original_token", 1234)
-    authenticator = _mock_august_authenticator()
-    token_refresh_lock = asyncio.Lock()
-
-    data = august.AugustData(
-        hass, MagicMock(name="api"), authentication, authenticator, token_refresh_lock
-    )
-    await data._async_refresh_access_token_if_needed()
-    authenticator.refresh_access_token.assert_not_called()
-
-    authenticator.should_refresh.return_value = 1
-    authenticator.refresh_access_token.return_value = _mock_august_authentication(
-        "new_token", 5678
+    binary_sensor_online_with_doorsense_name_open = hass.states.get(
+        "binary_sensor.online_with_doorsense_name_open"
     )
-    await data._async_refresh_access_token_if_needed()
-    authenticator.refresh_access_token.assert_called()
-    assert data._access_token == "new_token"
-    assert data._access_token_expires == 5678
-
-
-def _create_august_data_with_lock_details(lock_details):
-    locks = []
-    for lock in lock_details:
-        if isinstance(lock, LockDetail):
-            locks.append(_mock_august_lock(lock.device_id))
-    authentication = _mock_august_authentication("original_token", 1234)
-    authenticator = _mock_august_authenticator()
-    token_refresh_lock = MagicMock()
-    api = MagicMock()
-    api.get_lock_detail = MagicMock(side_effect=lock_details)
-    api.get_operable_locks = MagicMock(return_value=locks)
-    api.get_doorbells = MagicMock(return_value=[])
-    return august.AugustData(
-        MagicMock(), api, authentication, authenticator, token_refresh_lock
+    assert binary_sensor_online_with_doorsense_name_open.state == STATE_ON
+    binary_sensor_missing_doorsense_id_name_open = hass.states.get(
+        "binary_sensor.missing_doorsense_id_name_open"
     )
+    assert binary_sensor_missing_doorsense_id_name_open is None
+
+
+async def test_set_up_from_yaml(hass):
+    """Test to make sure config is imported from yaml."""
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    with patch(
+        "homeassistant.components.august.async_setup_august", return_value=True,
+    ) as mock_setup_august, patch(
+        "homeassistant.components.august.config_flow.AugustGateway.authenticate",
+        return_value=True,
+    ):
+        mocked_config = _mock_get_config()
+        assert await async_setup_component(hass, "august", mocked_config)
+    await hass.async_block_till_done()
+    assert len(mock_setup_august.mock_calls) == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5068548)